src/main.rs

```diff
@@ -62,22 +61,131 @@ fn main() -> anyhow::Result<()> {
         / (n - 1) as f64)
         .powf(0.5);
 
+    let (histogram_x, division_val_x) = histogram(&positions, |x| x.x, (avg, std_dev));
+    let (histogram_y, division_val_y) = histogram(&positions, |x| x.y, (avg, std_dev));
+    let (histogram_z, division_val_z) = histogram(&positions, |x| x.z, (avg, std_dev));
+
+    let histogram_val_x = {
+        let mut histogram_val = vec![0; division_val_x.len()];
+
+        for idx in histogram_x.iter().map(|(i, _)| *i as usize) {
+            histogram_val[idx] = histogram_val[idx] + 1;
+        }
+        histogram_val
+    };
+    let histogram_val_y = {
+        let mut histogram_val = vec![0; division_val_y.len()];
+
+        for idx in histogram_y.iter().map(|(i, _)| *i as usize) {
+            histogram_val[idx] = histogram_val[idx] + 1;
+        }
+        histogram_val
+    };
+    let histogram_val_z = {
+        let mut histogram_val = vec![0; division_val_z.len()];
+
+        for idx in histogram_z.iter().map(|(i, _)| *i as usize) {
+            histogram_val[idx] = histogram_val[idx] + 1;
```

Comment:
Same here. Plus you could collect all this logic into a function instead of repeating it for all 3 axes

Reply:
Fixed